codechain/primitives/AssetAddress.py

```diff
@@ -0,0 +1,110 @@
+import re
+from typing import List
+from typing import NamedTuple
+from typing import Tuple
+from typing import Union
+
+from ..crypto import bech32_decode
+from ..crypto import bech32_encode
+from .HexString import H160
+
+
+class MultiSig(NamedTuple):
+    n: int
+    m: int
+    pubkeys: List[H160]
+
+
+class AssetAddress:
+    def __init__(
+        self, address_type: int, payload: Union[H160, str, MultiSig], address: str
+    ):
+        self.address_type = address_type
+
+        if isinstance(payload, (H160, str)):
+            self.payload = H160(payload)
+        else:
+            n, m, pubkeys = payload
+            self.payload = (n, m, list(map(lambda p: H160(p), pubkeys)))
+        self.value = address
+
+    @classmethod
```

Comment:
Okay.

Reply:
But this said https://stackoverflow.com/questions/1950414/what-does-classmethod-do-in-this-code/1950927#1950927 one of the main uses of classmethod is to define alternative constructors.

Reply:
It might be useful when we let `AssetAddress` inheritable. It'll work as an alternative `__new__`, which gives you the `cls` parameter too. However, it doesn't seem to useful to inherit it. (by the way, if you want it to be inheritable, then you should prefer named parameters rather than positional arguments in constructors) In this case, `@classmethod` only leaves unnecessary access to `cls`.

Reply:
Okay.